### DISCUSSION TITLE
Xbox JIT/hardware limitations

### DISCUSSION THREAD

**Discussion by AshleyScirra** [MicrosoftEdge/WebView2Feedback#4120] [Q&A] [answered]
Thanks for launching Xbox support for WebView2! I have a couple of questions:

- Is the JavaScript JIT enabled for WebView2 on Xbox?
- Are there any hardware limitations on WebView2 on Xbox, such as limitations on the amount of CPU, GPU or RAM it's allowed to use?

**Comment by la-thorn** [comment]
I don't know the answer to your first question off the top of my head. Is there any more context? Do you have a reason to believe the JavaScript JIT might not be running on Xbox, or is it just out of curiosity?

As to the second question, apps using WebView2 are subject to the same hardware limitations as any other UWP running on Xbox, which is documented here:
https://learn.microsoft.com/en-us/windows/uwp/xbox-apps/system-resource-allocation

**Reply by AshleyScirra** [reply]
Some console platforms have historically restricted dynamic code generation, which would rule out running a JIT, and therefore reduce performance. Given this history of disabling JIT on console platforms I was curious as to whether WebView2 is able to make use of JIT on Xbox.

**Reply by la-thorn** [reply, marked as answer]
I did a little digging; the JavaScript JIT compiler does indeed run in WebView2 on Xbox.

**Reply by AshleyScirra** [reply]
Great, thanks for confirming!
